app/controllers/councillor_contributions_controller.rb

```diff
@@ -17,6 +17,23 @@ def new
     @councillor_contribution.suggested_councillors.build({email: nil, name: nil}) if new_suggested_councillor_required?
   end
 
+  def source
+    @authority = Authority.find_by_short_name_encoded!(params[:authority_id])
+
+    @councillor_contribution = @authority.councillor_contributions.build(
+      councillor_contribution_with_suggested_councillors_params
+    )
+
+    if @councillor_contribution.suggested_councillors.empty?
+      @councillor_contribution.suggested_councillors.build({email: nil, name: nil})
+    end
+
+    unless @councillor_contribution.valid?
```

Comment:
Given our discussion here @hisayohorie https://github.com/openaustralia/planningalerts/pull/1246#issuecomment-327966708 should we be saving the contribution at this point instead of just validating, in case the person has to abandon ship at this point?

Reply:
@equivalentideas I gave some thoughts about it and now I'm not sure. Maybe because the way I set it up it feels like the submission is in the middle.

Reply:
> @equivalentideas I gave some thoughts about it and now I'm not sure. Maybe because the way I set it up it feels like the submission is in the middle.

@hisayohorie and I spoke about this on a call last week and decided that the safest option at the moment is to save the contribution on when the councillors are submitted, before adding the source and contributor.

There are many reasons why someone might halt the process before hitting 'submit' on the 'add source' page (e.g. they get distracted by kids, laptop runs out of battery) and we wouldn't want to have wasted their time by throwing away all their work.

I'll implement that change now 😄